internal/addon/secrets.go

```diff
@@ -9,33 +9,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// Endpoint defines the name of an endpoint that will be available to store
-// signal data.
-type Endpoint string
-
 // GetSecrets fetches Kubernetes secrets based on the specified
 // secret name for each target in targetSecretName.
 // If a secret doesn't exist in the namespace of the ManagedClusterAddon this
 // function will instead look for it in the configResourceNamespace.
 // If no secret is found an error will be returned.
-func GetSecrets(ctx context.Context, k8s client.Client, configResourceNamespace string, addonNamespace string, targetSecretName map[Endpoint]string) (map[Endpoint]corev1.Secret, error) {
-	secretKeys := make(map[Endpoint]corev1.Secret, len(targetSecretName))
-	for targetName, secretName := range targetSecretName {
-		secretReference := &corev1.Secret{}
+func GetSecrets(ctx context.Context, k8s client.Client, configResourceNamespace string, addonNamespace string, secretNames []string) ([]corev1.Secret, error) {
+	secrets := make([]corev1.Secret, 0, len(secretNames))
```

Comment:
nit. I think we don't buy much with pre-allocation here, don't we? It's obscure for such small slices anyway.